lib/appsignal/probes/mri.rb

```diff
@@ -34,7 +31,8 @@ def call
         )
 
         set_gauge("thread_count", Thread.list.size)
-        set_gauge("gc_total_time", MriProbe.garbage_collection_profiler.total_time)
+        total_time = gauge_delta(:gc_total_time, @gc_profiler.total_time)
+        set_gauge("gc_total_time", total_time) if total_time
```

Comment:
Wait a minute. If the value is continuously growing, and it will reset to zero every 23 days, which I assume that is when it overflows the integer or whatever... then wouldn't that mean that `gauge_delta` [(which calculates current value minus previous value)](https://github.com/appsignal/appsignal-ruby/blob/61a78fb028b04ae6f0a4ca1fc469d744f23c5029/lib/appsignal/probes/helpers.rb#L25) will report a value of almost `INT_MIN` every 23 days? That probably makes our graph do something funny.

Luckily, since we know the value is always increasing, this should do. It will skip reporting a value every 23 days instead, which is preferable to breaking the graph.

```suggestion
        set_gauge("gc_total_time", total_time) if total_time && total_time >= 0
```

Reply:
You're right! If we don't check if the value is larger than 0 it would, once every 23 days, report a very large negative value. I'll add this exception.

Reply:
@unflxw I've added a new test and helper to check this scenario. Let me know what you think.